src/http-client/fetch-client.ts

```diff
@@ -2,25 +2,39 @@
 /// <reference lib="dom" />
 
 import { NetworkError } from "../errors";
-import { HTTPClient, HTTPRequest, HTTPResponse } from "./index";
+import {
+  HTTPClient,
+  HTTPClientOptions,
+  HTTPRequest,
+  HTTPResponse,
+} from "./http-client";
 
 /**
  * An implementation for {@link HTTPClient} that uses the native fetch API
  */
 export class FetchClient implements HTTPClient {
+  #url: string;
+
+  constructor({ url }: HTTPClientOptions) {
+    this.#url = new URL("/query/1", url).toString();
```

Comment:
not a big fan of hiding the URL we actually use in the HTTP client layer but we can revisit later.